x/stakeibc/keeper/lsm.go

```diff
@@ -1,12 +1,55 @@
 package keeper
 
 import (
+	"time"
+
+	errorsmod "cosmossdk.io/errors"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 
+	"github.com/golang/protobuf/proto" //nolint:staticcheck
+
 	"github.com/Stride-Labs/stride/v8/x/stakeibc/types"
 )
 
-func (k Keeper) TransferLSMTokenToHost(ctx sdk.Context, lsmTokenDeposit types.LSMTokenDeposit) error {
-	// TODO [LSM]
+// Submits an ICA to "Redeem" an LSM Token - meaning converting the token into native stake
+// This function is called in the EndBlocker which means if the ICA submission fails,
+//   any modified state is not reverted
+// The deposit Status is intentionally updated before the ICA is submitted even though it will NOT be reverted
+//   if the ICA fails to send. This is because a failure is likely caused by a closed ICA channel, and the status
```

Comment:
From the logic in `DetokenizeCallback`, it seems like the case you're designing around here is timeout: if timeout, leave the status as `DETOKENIZATION_IN_PROGRESS`. 

Can timeouts only occur if the channel closes / is closed, since it's an ordered channel?

In that case, you say `When the ICA channel is restored, the deposit status will get reset, and the ICA will be attempted again.`. Where's the logic to reset the status when the channel re-opens?

Reply:
> Can timeouts only occur if the channel closes / is closed, since it's an ordered channel?
its more that IF a timeout occurs, then the channel gets closed automatically

So an example is:
* ICA #1 is submitted, channel gets closed
* ICA #2 attempted immediately after, but SubmitTxs fails since the channel is closed

Since this is all triggered from the begin blocker, if we revert the record right away, the begin blocker will try to submit it again in the next block. But I think we actually just want to ignore the record until the channel is restored
Fwiw, there's also a clause in the being blocker that only submits this ICA if the channel is open, so I'm actually not even sure this case is even relevant. But in any event, it felt like it made more sense to update the status rather than infinitely retry

> Where's the logic to reset the status when the channel re-opens?
And there will be a restore-channel transaction that reverts all IN_PROGRESS records to QUEUE

Reply:
You said a restore-channel transaction would revert all IN_PROGRESS to QUEUE but is that known to be safe? Are there guarantees on how long the detokenization will take so some from previous blocks which really are being detokenized can't still be IN_PROGRESS at the time when a channel gets closed?  It seems like you want the ones which were just moved to IN_PROGRESS only and not all which are that status unless these can be guaranteed to be the same thing.

Reply:
You are saying a completely new channel needs to be made and any/all in flight requests would need to be resent... how does this work on the hubside?  So let;s say there is a detokenize in process so it was recieved hubside and it is doing it, but the channel closes.  Will the detokenize fail because it is unable to callback?

Reply:
when the channel closes, it closes on both ends. The important part here is the ordering. If you send out 5 packets back to back, the last 4 cannot be submitted to the hub until the first packet was either an ack success or ack failure.
And if packet 1 times out, then the channel closes on both the hub and Stride and packets 2-5 can never be submitted to the hub